workflow-templates/im-test-sam-yaml.yml

```diff
@@ -29,7 +29,7 @@ on:
 jobs:
   sam-yaml:
     name: Validate and Export SAM YAML
-    runs-on: [ubuntu-latest]
+    runs-on: [self-hosted, ubuntu-20.04]
```

Comment:
What's the thinking for changing this to our self-hosted runners?

Reply:
It's in the PR description. 😄
> Because the stewardship export is on by default, this must run on a self-hosted runner, otherwise the json upload will come from the public internet and will be blocked by Cosmos IP restrictions.

Reply:
You can see an example of this failure here: https://github.com/im-client/help-and-support/runs/8025578592?check_suite_focus=true#step:4:8

Reply:
It's true the firewall protection rules won't let machines from "off-prem" or unknown networks write to it now.  I recently had to add it to the cosmos account because of a policy that's going to be enforced mid Sept.

Reply:
@hpractv you might need to take those out or greenlight the current github ips so the existing workflows don't break until we get a chance to update it.